apps/rule-manager/app/service/RuleManager.scala

```diff
@@ -416,33 +416,55 @@ object RuleManager extends Loggable {
 
   def destructivelyPublishDictionaryRules(
       words: List[String],
-      bucketRuleResource: BucketRuleResource
+      bucketRuleResource: BucketRuleResource,
+      wordsToNotPublish: List[WordTag]
   ) = {
     // Destroy existing draft dictionary rules
     DbRuleDraft.destroyDictionaryRules()
     // Destroy existing live dictionary rules
     DbRuleLive.destroyDictionaryRules()
+    var availableTags = Tags.findAll()
+    // Add any tags from wordsToNotPublish that aren't already in the DB
+    val tagsToAdd = wordsToNotPublish
+      .map(wordTag => wordTag.tag)
+      .distinct
+      .filter(tagName => !availableTags.exists(tag => tag.name == tagName))
+      .map(name => Tag(None, name))
+    if (tagsToAdd.size > 0) {
+      Tags.batchInsert(tagsToAdd)
+      availableTags = Tags.findAll()
+    }
 
     val initialRuleOrder = DbRuleDraft.getLatestRuleOrder() + 1
-    val dictionaryRules = words.zipWithIndex.map(wordAndIndex =>
-      DbRuleDraft.withUser(
-        id = None,
-        ruleType = "dictionary",
-        pattern = Some(wordAndIndex._1),
-        category = Some("Collins Dictionary"),
-        ignore = false,
-        user = "Collins Dictionary",
-        ruleOrder = initialRuleOrder + wordAndIndex._2,
-        externalId = None
+    val dictionaryRules = words.zipWithIndex
+      .map(word => {
+        val tagId = wordsToNotPublish.find(taggedWord => taggedWord.word == word) match {
```

Comment:
Does this definitely do what's expected? The lhs is `Wordtag` and the rhs is `(String, Int)`

Reply:
Discussed out of band - it wasn't working.